gadgets/clock/src/lib.rs

```diff
@@ -17,3 +13,19 @@ impl Gadget for ClockGadget {
         });
     }
 }
+
+pub struct ClockGadgetFactory;
```

Comment:
You have the order `Gadget` `impl Gadget` `GadgetFactory` `impl GadgetFactory` here but `Gadget` `GadgetFactory` `impl Gadget` `impl GadgetFactory` for the other two.

Please choose one of the two and enforce it, so we get a consistent style across gadgets

Reply:
Ah that's a mistake, the style I'm trying to tend to is "struct definitions first, then impls, then free functions", will fix soon